addons/dogtags/script_component.hpp

```diff
@@ -0,0 +1,1029 @@
+#define COMPONENT dogtags
+#include "\z\ace\addons\main\script_mod.hpp"
+
+// #define DEBUG_MODE_FULL
+// #define DISABLE_COMPILE_CACHE
+// #define CBA_DEBUG_SYNCHRONOUS
+// #define ENABLE_PERFORMANCE_COUNTERS
+
+#ifdef DEBUG_ENABLED_DOGTAGS
+    #define DEBUG_MODE_FULL
+#endif
+
+#ifdef DEBUG_SETTINGS_DOGTAGS
+    #define DEBUG_SETTINGS DEBUG_SETTINGS_DOGTAGS
+#endif
+
+#include "\z\ace\addons\main\script_macros.hpp"
+
+#define ACE_DOGTAG_CLASS(DOGTAGBASE,DOGTAGID) \
+    class DOGTAGBASE##_##DOGTAGID : DOGTAGBASE { \
+        author = ECSTRING(common,ACETeam); \
+        scope = 1; \
+        scopeArsenal = 0; \
+        scopeCurator = 0; \
+        descriptionShort = QUOTE(DOGTAGID); \
+        GVAR(tagID) = DOGTAGID; \
+    };
+
+#define ACE_DOGTAGS(DOGTAGBASE) \
+    ACE_DOGTAG_CLASS(DOGTAGBASE,1) \
```

Comment:
pretty sure you can make a loop with preprocessor commands, so this 1000 item list isn't necessary.